test/e2e/e2e_test.go

```diff
@@ -38,6 +40,128 @@ import (
+
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteLocalQueue(ctx, k8sClient, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(util.DeleteAllJobsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, defaultRF, true)
+		})
+
+		ginkgo.It("Should fetch information about pending workloads", func() {
+			ginkgo.By("Have zero pending workloads", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, clusterQueue.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(0))
+			})
+
+			ginkgo.By("Schedule two max capacity jobs", func() {
+				highPriorityClass := testing.MakePriorityClass("high").PriorityValue(100).Obj()
+				gomega.Expect(k8sClient.Create(ctx, highPriorityClass))
+				ginkgo.DeferCleanup(func() {
+					gomega.Expect(k8sClient.Delete(ctx, highPriorityClass)).To(gomega.Succeed())
+				})
+
+				lowPriorityClass := testing.MakePriorityClass("low").PriorityValue(50).Obj()
+				gomega.Expect(k8sClient.Create(ctx, lowPriorityClass))
+				ginkgo.DeferCleanup(func() {
+					gomega.Expect(k8sClient.Delete(ctx, lowPriorityClass)).To(gomega.Succeed())
+				})
+
+				sampleJob1 = testingjob.MakeJob("test-job-1", ns.Name).
+					Queue(localQueue.Name).
+					Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"60s", "-termination-grace-period", "0s"}).
+					Request(corev1.ResourceCPU, "1").
+					PriorityClass(highPriorityClass.Name).
+					Obj()
+				sampleJob2 = testingjob.MakeJob("test-job-2", ns.Name).
+					Queue(localQueue.Name).
+					Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1ms"}).
+					Request(corev1.ResourceCPU, "1").
+					PriorityClass(lowPriorityClass.Name).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, sampleJob1)).Should(gomega.Succeed())
+				gomega.Expect(k8sClient.Create(ctx, sampleJob2)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Have one pending workload", func() {
+				gomega.Eventually(func() int {
+					info, err := visibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, clusterQueue.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(1))
+			})
+
+			ginkgo.By("Await for pods to be running", func() {
+				gomega.Eventually(func() int {
+					createdJob := &batchv1.Job{}
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(sampleJob1), createdJob)).Should(gomega.Succeed())
+					return int(*createdJob.Status.Ready)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(1))
+				// Make sure that pod can receive SIGTERM
+				time.Sleep(time.Second * 1)
```

Comment:
Wrap this sleep with `gomega.Expect(util.DeleteAllPodsInNamespace(ctx, k8sClient, ns)).Should(gomega.Succeed())` into a single step `Terminate execution of the first workload to release the quota`

Reply:
We tried to fix it without sleep, and send Delete (SIGTERM) to the running pod, but it some case the pod terminated only after 20-30s (probably due to SIGKILL).

We think so, when repeated 50times, we got error rate about 10%, but haven't pinned down the root cause exactly.

Reply:
I think it is worth to dig deeper and don't use sleep, however, it is used already in some pre-existing tests, so maybe it is acceptable, to use the sleep and open a follow up issue to solve it properly?

Reply:
It would probably require a readiness probe: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#types-of-probe

But it sounds like an overkill.

Still deferring the LGTM of this PR to you, @mimowo

Reply:
I'm fine with the 5s delay for now, as long as we create a issue to follow up.

Reply:
I think 1s delay should be enough. We are only calling Delete once the Job reports `status.ready=1`. This would only happen if the Pod is already running, as observed by kubelet.

Reply:
I see, so you are suggesting to use `DeleteAllPodsRunning`, but prior to this do `status.ready=1` + wait for a second. I like this more that 5s, @PBundyra tested and it passes 50/50. @PBundyra could you revert to this approach then?

Reply:
Unfortunately, after reverting the tests flake: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1251/pull-kueue-test-e2e-main-1-25/1727366389850705920. Sorry for going back and forth, but maybe return to the 5s delay (without delete), and ticket a better solution. We already have tests waiting 5s.

Reply:
@alculquicondor WDYT?

Reply:
surprising how long it can take for a Pod to run through a few system calls :(
Still worth waiting for `status.ready==1`, and maybe a 3s sleep?

Reply:
We are just making sure the pod is scheduled, reducing the risk of the pod not being running after 5s

Reply:
To update the thread, we go with the approach of terminating the job1 one from the test code. In order to make the test not flake we also set `backoffLimit=0` and `terminationGracePeriodSeconds=1`. The job can either fail or succeed, but it is not relevant for the test, because we just want to unblock capacity.